File: RssTokens.rb
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# RssTokens.rb

require "./TokenTypes.rb"

class RssTokens < TokenTypes
    public
        def initialize
            super()
        end
        
        def set()
            @types["Token"] = 0
            @types["Url"] = 1
            @types["VersionNum"] = 2
            @types["="] = 3
            @types["\""] = 4
            @types["<"] = 5
            @types[">"] = 6
            @types["</"] = 7
            @types["version"] = 8
            @types["rss"] = 9
            @types["channel"] = 10
            @types["title"] = 11
            @types["link"] = 12
            @types["description"] = 13
            @types["item"] = 14
            @types["Error"] = 15
            @types["InputEnd"] = 16
        end
        
        def isKeyword?( symbol )
            keywords = ["version", "rss", "channel", "title", "link", "description", "item"]
            
            return ( keywords.index( symbol ) != nil )
        end
end

File: RssTokens.py
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# RssTokens.py

import TokenTypes

class RssTokens( TokenTypes.TokenTypes ):

    def __init__( self ):
        super( RssTokens, self ).__init__()
        
    def setTypes( self ):
        self.types["Token"] = 0
        self.types["Url"] = 1
        self.types["VersionNum"] = 2
        self.types["="] = 3
        self.types["\""] = 4
        self.types["<"] = 5
        self.types[">"] = 6
        self.types["</"] = 7
        self.types["version"] = 8
        self.types["rss"] = 9
        self.types["channel"] = 10
        self.types["title"] = 11
        self.types["link"] = 12
        self.types["description"] = 13
        self.types["item"] = 14
        self.types["Error"] = 15
        self.types["InputEnd"] = 16
        
    def isKeyword( self, symbol ):
        keywords = ["version", "rss", "channel", "title", "link", "description", "item"]
        
        return symbol in keywords
        
